agent/consul/server.go

```diff
@@ -1296,6 +1304,22 @@ func (s *Server) RPC(method string, args interface{}, reply interface{}) error {
 		reply:  reply,
 	}
 
+	ctx := context.Background()
+	tracer := otel.GetTracerProvider().Tracer("consul/server")
+	recorder := RequestRecorder{
+		Logger:      s.logger.Named("rpc/recorder"),
+		Tracer:      tracer,
+		SpanOptions: []trace.SpanStartOption{trace.WithSpanKind(trace.SpanKindServer)},
+	}
+	if v, ok := args.(structs.RPCInfo); ok {
+		ctx = structs.Extract(ctx, v)
+	}
+	ctx, span := tracer.Start(ctx, "Consul Internal RPC", recorder.SpanOptions...)
+	defer span.End()
+	span.SetAttributes(
+		IsLocalAttribute.Bool(true),
+	)
```

Comment:
Internal RPC's are supposed allow the Consul Server to make RPC's against itself without hitting the network. The Span's tell me it's still hitting the net/rpc server layer